### ISSUE TITLE
Switch log is empty

### ISSUE THREAD

**Issue by reinhold-willcox** [TeamCOMPAS/COMPAS#870]
**Describe the bug**
First identified by @solfreludio. When the switch log is enabled (in SSE or BSE), it creates an empty directory. None of the stellar type switches are correctly printed. 

This issue was introduced in v2.32.00, when the logfile record types were added, so I'm assigning @jeffriley. 

**Label the issue**
`urgency_low`      - This issue is not urgent
`severity_minor`    - This is a minor bug with minimal impact

**To Reproduce**
```COMPAS --switch-log ```

**Expected behavior**
Stellar type switches should be printed.

**Version**
 - COMPAS v02.34.00

**Comment by nrsegovia** [comment]
I'm far from being an expert on the way that the logs have been implemented, but tracing the issue led me to line 993 in Log.h, the condition `if (((1 << (p_RecordType - 1)) & fileDetails.recordTypes) > 0)` ** fails for the switch logs (tested on SSE, should be the same for BSE). There are equivalent lines, one specific to HDF5 and another for different file formats.

Furthermore, it seems that this is related to `p_RecordType` being equal to `0U` when `LogStandardRecord` is called through `LogSSESwitchLog` (line 1214, same file). Changing it to `1U` makes the switch logs work again in SSE, and I believe the fix should be equivalent for BSE. However, I do not know if this has a wider impact in the code.

** = I think the one that fails might instead be `if ((p_RecordType & fileDetails.recordTypes) > 0)`, which is included in the `LogStandardRecord` functions that are used after records are constructed. Evaluation of `p_RecordType & fileDetails.recordTypes` using `p_RecordType = 0U` and `fileDetails.recordTypes = -1` (which is what happens with the switch logs, as per what I was able to trace back) gives 0, making the condition fail.

**Comment by jeffriley** [comment]
I'll take a look at this today or tomorrow.  I thought I allowed for the switch logs when I implemented record types, but it looks like I may have either done that badly or missed something...

**Comment by jeffriley** [comment]
@solfreludio is right - the p_RecordType parameter passed to LogStandardRecord() for the switchlog files should be non-zero (any positive integer would work, so 1 is a good choice).  

The p_RecordType parameter passed to LogStandardRecord() is used to shift the integer 1 left to match a bitmap of the record types to be printed.  Record type aren't used for the switch logs, so really the parameter doesn't matter for the switch logs - except that we subtract 1 from the parameter passed ans then shift by the resulting number of bits.  Shifting left (or right) by a negative number results in undefined behaviour...  My mistake - since the parameter doesn't really matter I just used a 0 - I should have used a 1 (or any positive integer).  I thought I tested switchlogs when I added the record type code, but I either didn't, or when I did the "undefined behaviour" worked in my favour (or not, depending upon how you look at it).

Anyway, the fix is to pass 1U to LogStandardRecord() for the switchlogs instead of 0U.  I'll try to push a fix later today (or maybe tomorrow).
